ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -1152,42 +1147,62 @@ private void encryptChunk() {
         logger.trace("Submitting encrypt job for chunk at index {}", chunkIndex);
         cryptoJobHandler.submitJob(
             new EncryptJob(passedInBlobProperties.getAccountId(), passedInBlobProperties.getContainerId(),
-                isMetadataChunk() ? null : buf, ByteBuffer.wrap(chunkUserMetadata), kms.getRandomKey(), cryptoService,
-                kms, encryptJobMetricsTracker, (EncryptJob.EncryptJobResult result, Exception exception) -> {
-              logger.trace("Processing encrypt job callback for chunk at index {}", chunkIndex);
-              encryptJobMetricsTracker.onJobResultProcessingStart();
-              if (exception == null && !isOperationComplete()) {
-                if (!isMetadataChunk()) {
-                  buf = result.getEncryptedBlobContent();
-                }
-                encryptedPerBlobKey = result.getEncryptedKey();
-                chunkUserMetadata = result.getEncryptedUserMetadata().array();
-                logger.trace("Completing encrypt job result for chunk at index {}", chunkIndex);
-                prepareForSending();
-                chunkReadyAtMs = time.milliseconds();
-              } else {
-                encryptJobMetricsTracker.incrementOperationError();
-                if (!isOperationComplete()) {
-                  logger.trace("Setting exception from encrypt of chunk at index {} ", chunkIndex, exception);
-                  setOperationExceptionAndComplete(
-                      new RouterException("Exception thrown on encrypting the content for chunk at index " + chunkIndex,
-                          exception, RouterErrorCode.UnexpectedInternalError));
-                } else {
-                  logger.trace(
-                      "Ignoring exception from encrypt job for chunk at index {} as operation exception {} is set already",
-                      chunkIndex, getOperationException(), exception);
-                }
-              }
-              routerMetrics.encryptTimeMs.update(time.milliseconds() - chunkEncryptReadyAtMs);
-              encryptJobMetricsTracker.onJobResultProcessingComplete();
-              routerCallback.onPollReady();
-            }));
+                isMetadataChunk() ? null : toEncrypt.retainedDuplicate(), ByteBuffer.wrap(chunkUserMetadata),
+                kms.getRandomKey(), cryptoService, kms, encryptJobMetricsTracker,
+                (EncryptJob.EncryptJobResult result, Exception exception) -> {
```

Comment:
This lambda is quite long. Could you move it to a helper function. e.g. `void onEncryptionComplete(EncryptJobResult, Exception)`?